### ISSUE TITLE
Can I help you with mod?

### ISSUE THREAD

**Issue by ErionError** [zippity21/Stardew_Valley_Regression_Mod#61]
Hi I'm just wandering if I can help you to finish mod.
I'm bad at Stardew Valley modding (Anyway I know c# .-.), I can help you with art at least. 
I rely want to finish this mod. (at least add everything from Issues).
For now, I did redraw the sprites. And planning to improve them for better look (add some wet-patches instead color fill. But it will affect on fullness indicator, so it will be better to add some other indication .-. or rework current)

![sprites](https://user-images.githubusercontent.com/110569190/182713394-6f575c96-ff79-4443-9fd4-7483d1ec4a31.png)
current reworked sprites version.

**Comment by zippity21** [comment]
> Hi I'm just wandering if I can help you to finish mod. I'm bad at Stardew Valley modding (Anyway I know c# .-.), I can help you with art at least. I rely want to finish this mod. (at least add everything from Issues). For now, I did redraw the sprites. And planning to improve them for better look (add some wet-patches instead color fill. But it will affect on fullness indicator, so it will be better to add some other indication .-. or rework current)
> 
> ![sprites](https://user-images.githubusercontent.com/110569190/182713394-6f575c96-ff79-4443-9fd4-7483d1ec4a31.png) current reworked sprites version.

Absolutely! One of the great things about Git is that you can pull the source and modify it to your heart's content without worrying about breaking it for somebody else.

If you do make the modifications and are happy with the results, you can do a pull request to have it added to the main release.

I'm willing to help however I can, so feel free to reach out as you work.

**Comment by ErionError** [comment]
Heh, that's really hard to read:
`int num = type != null ? (!(type == "drying") ? (!(type == "messy") ? (!(type == "wet") ? LARGE_SPRITE_DIM*2 : LARGE_SPRITE_DIM) : LARGE_SPRITE_DIM*2) : LARGE_SPRITE_DIM*3) : (!c.IsDrying() ? ((double)c.messiness <= 0.0 ? ((double)c.wetness <= 0.0 ? 0 : LARGE_SPRITE_DIM) : LARGE_SPRITE_DIM*2) : LARGE_SPRITE_DIM*3);`
But I suppose that's code that slices the sprite sheet?

**Comment by ErionError** [comment]
![sprites](https://user-images.githubusercontent.com/110569190/182921201-56de5c24-47be-4ab2-b2a7-96dc11259d9e.png)
I updated textures so they have more pleasant look when dirty. But I discovered bug with underwear icon, both sprites overlap.
Just interesting, are you need some additional sprites? (something like: toilet, or TV screens)

**Comment by zippity21** [comment]
> ![sprites](https://user-images.githubusercontent.com/110569190/182921201-56de5c24-47be-4ab2-b2a7-96dc11259d9e.png) I updated textures so they have more pleasant look when dirty. But I discovered bug with underwear icon, both sprites overlap. Just interesting, are you need some additional sprites? (something like: toilet, or TV screens)

Based on the current wetness/messiness indicator, the overlap is intentional. The way it currently works is that the "dry" image is always present, with the wet and messy overlaid, but only displaying a portion of the image based on percent used.

To use separate sprites is fine (I like the designs), but that code will need to be changed.

**Comment by ErionError** [comment]
> 

You mean more variation of fullness?

**Comment by zippity21** [comment]
> > More sprites would be great, but there is currently no mechanism to use them (though they could be made for future use).
> 
> You mean more variation of fullness?

I was referencing your idea for toilet sprite etc.

I don't necessarily think the fullness indicator is necessary since you can hover over it for numbers. So going with static images like you made should work.

**Comment by ErionError** [comment]
> I was referencing your idea for toilet sprite etc.

Yes, I didn't notice.

> I don't necessarily think the fullness indicator is necessary since you can hover over it for numbers. So going with static images like you made should work.

Hmm... That makes sense. I will try to dig up your code and maybe fix this issue.

**Comment by zippity21** [comment]
> > I was referencing your idea for toilet sprite etc.
> 
> Yes, I didn't notice.
> 
> > I don't necessarily think the fullness indicator is necessary since you can hover over it for numbers. So going with static images like you made should work.
> 
> Hmm... That makes sense. I will try to dig up your code and maybe fix this issue.

Also, a my suggestion would be, that if you go the route of static images, you'll probably want to add a row of sprites for when you're both wet and messy.

**Comment by ErionError** [comment]
> > > I was referencing your idea for toilet sprite etc.
> > 
> > 
> > Yes, I didn't notice.
> > > I don't necessarily think the fullness indicator is necessary since you can hover over it for numbers. So going with static images like you made should work.
> > 
> > 
> > Hmm... That makes sense. I will try to dig up your code and maybe fix this issue.
> 
> Also, a my suggestion would be, that if you go the route of static images, you'll probably want to add a row of sprites for when you're both wet and messy.

Yes. I thought about it, but for the current system, I decided to make the possibility for them to overlap and make something like a sandwich. However, with a new system, it will make sense to do another row.

**Comment by ErionError** [comment]
So. I did this. It looks better now and does not overlap. I also add a fifth wet end mess texture.
![StardewModdingAPI_xHx7Jh4mMv](https://user-images.githubusercontent.com/110569190/183148438-674efbab-3931-4296-ba56-695a069e489e.png)
That's hoe it look.

I also changed the underwear change button to a left mouse.

**Comment by ErionError** [comment]
Just wandering, did you know how to add new slot for underwear?

**Comment by zippity21** [comment]
> Just wandering, do you know how to add new slot for underwear?

As in adding an equipment slot? If so, then no, but there are other mods that do add equipment slots, so you could probably look to those for inspiration.

Also, I suggest working on one feature at a time. Doing too many things in a single pull request will make reviewing it much more difficult.
